python/lsst/ip/diffim/subtractImages.py

```diff
@@ -507,8 +512,17 @@ def finalize(self, template, science, difference, kernel,
         mask = difference.mask
         mask &= ~(mask.getPlaneBitMask("DETECTED") | mask.getPlaneBitMask("DETECTED_NEGATIVE"))
 
+        # We have cleared the template mask plane, so copy the mask plane of
+        # the image difference so that we can calculate correct statistics
+        # during decorrelation. Do this regardless of whether decorrelation is
+        # used for consistency.
+        template[science.getBBox()].mask.array[...] = difference.mask.array[...]
         if self.config.doDecorrelation:
             self.log.info("Decorrelating image difference.")
+            # We have cleared the template mask plane, so copy the mask plane of
+            # the image difference so that we can calculate correct statistics
+            # during decorrelation
+            template[science.getBBox()].mask.array[...] = difference.mask.array[...]
```

Comment:
I don't understand why this line is outside the if and also inside the if statement. I wouldn't place it in the if. Looking at the inline comments, I would think that the correct thing is to do just with line 519.

Reply:
Good catch! This line is indeed a duplicate of 519. I'll remove it.